### PR TITLE
transform: (preprocess-mlperftiny) initialize preprocessing pass

### PR DESCRIPTION
**Pull request by jorendumoulin**
This PR introduces the preprocessing pass for mlperftiny networks.
It is a collection of lowering patterns and ad-hoc transformations to get the network into the nice format to be ingested by the compiler.
Perhaps with a better frontend for neural networks, this would be slightly less ugly, but this performs quite nicely for now.

This awaits #243, who's changes are already present here.

### CODE REVIEW

File: compiler/transforms/frontend/preprocess_mlperf_tiny.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+        if not isinstance((const_op := const_input.op), arith.Constant):
+            return
+        if not isinstance((intattr := const_op.value), builtin.IntegerAttr):
+            return
+        if intattr.value.data != 0:
+            return
+
+        # erase the op
+        rewriter._replace_all_uses_with(op.results[0], op.outputs[0])
+        rewriter.erase_matched_op()
```
Comment (JosseVanDelm) [inline compiler/transforms/frontend/preprocess_mlperf_tiny.py:140]: I guess the difference w.r.t `InsertMemoryClears` is that this has to do with initialization, whereas that has to do more with freeing/allocation?

**Review by JosseVanDelm — APPROVED** [review]
Nice work Joren!

File: compiler/transforms/convert_tosa_to_kernel.py [hunk 2]
```diff
@@ -51,8 +52,12 @@ def match_and_rewrite(self, rescale_op: tosa.RescaleOp, rewriter: PatternRewrite
         multiplier = rescale_op.multiplier.data.data[0].data
         assert isinstance(multiplier, int)
         shift = assert_int8(rescale_op.shift.data.data[0].data)
-        max_int = assert_int8(clamp_op.max_int.value.data)
-        min_int = assert_int8(clamp_op.min_int.value.data)
+        if isinstance(clamp_op, tosa.ClampOp):
+            max_int = assert_int8(clamp_op.max_int.value.data)
+            min_int = assert_int8(clamp_op.min_int.value.data)
+        else:
```
[the remark this answers is not in the document]
Comment (jorendumoulin) [reply compiler/transforms/convert_tosa_to_kernel.py:58]: clamping is performed either by rescaleop or by  clampop. if it is done by clampop, we take those values, if it is done by rescaleop, it is done with -128 and 127

File: compiler/transforms/frontend/preprocess_mlperf_tiny.py (hunk 1 above)
Comment (jorendumoulin) [reply compiler/transforms/frontend/preprocess_mlperf_tiny.py:140]: Yes!

**Pull request merged**